scrapers/twitter.js

```diff
@@ -1,34 +1,115 @@
+"use strict";
 var BaseLoklakScrapper = require('./base');
 
+const cheerio = require('cheerio');
+const request = require('request-promise-native');
+
 class TwitterLoklakScrapper extends BaseLoklakScrapper {
 
 	constructor() {
-		super('Twitter', 'https://twitter.com/');
+		super('Twitter', 'https://twitter.com/search?f=tweets&vertical=default&q=');
 	}
 
 	argumentSanityCheck(args) {
 		super.argumentSanityCheck(args);
-
-		if (args.length <= 2) {
-			console.error('Atleast one argument required.');
-			process.exit(-1);
-		}
-
 		return true;
 	}
 
 	onInit() {
-		this.REQUEST_URL = this.BASE_URL + '';
-		this.request();
+		this.REQUEST_URL = this.BASE_URL + this.SLICED_PROC_ARGS + "&src=typd";
 	}
 
 	scrape($) {
 		// Scraping logic goes here.
 		// We have our HTML and $ set here so we can directly use these.
 		// Set the result as this.JSON = {}
+		const tweets = [];
+		$("div.tweet").each( (i, element) => {
+			let tweetDetail = {};
+
+			// userid, avatarurl, fullname, nickname
+			tweetDetail.userid = $(element).attr("data-user-id");
+			tweetDetail.useravatarurl = $(element).find("img.avatar").attr("src");
+			tweetDetail.userfullname = $(element).attr("data-name");
+			tweetDetail.usernickname = $(element).attr("data-screen-name");
+
+			// status_url, time-name, time_in_millis
+			tweetDetail.tweetstatusurl = $(element).find("a.tweet-timestamp").attr("href");
+			tweetDetail.tweettimename = $(element).find("a.tweet-timestamp").attr("title");
+			tweetDetail.tweettimems = $(element).find("span._timestamp").attr("data-time-ms");
+
+			// tweet_text
+			tweetDetail.tweettext = $(element).find("p.tweet-text").text();
+
+			// TODO: parsing_retweet, parsing_favourite
+			$(element).find("span.ProfileTweet-actionCountForPresentation").each((i, ele) => {
+				let number = $(ele).text();
+				if (i === 1) {
+					if (number.length > 0) {
+						tweetDetail.tweetretweetcount = parseInt(number);
+					} else {
+						tweetDetail.tweetretweetcount = 0;
+					}
+				}
+				if (i === 3) {
+					if (number.length > 0) {
+						tweetDetail.tweetfavouritecount = parseInt(number);
+					} else {
+						tweetDetail.tweetfavouritecount = 0;
+					}
+				}
+			});
+
+			// tweet images
+			tweetDetail.images = []
+			$(element).find("img").each((i, ele) => {
+				let imageLink = $(ele).attr("src");
+				tweetDetail.images.push(imageLink);
+			});
+			tweetDetail.images.splice(0, 1); // removes the profile image url
+
+			// TODO: add location data
+			// TODO: video, audio, gifs
+			tweets.push(tweetDetail);
+		});
+		
+		tweets.splice(-1, 1); // all data is not available for the last tweet
```

Comment:
For last tweet only the first line i.e `<div class="tweet ...` is fetched by the GET request. There is no information available about the tweet at all (every attribute of the JSONObject is null). If we could get status url, then it other details can be scraped, but that isn't the case.